modules/serverless-rn-1-17-0.adoc

```diff
@@ -0,0 +1,60 @@
+
+* Changes to API versions in Knative Eventing 0.23.0:
+
+** Version `v1beta1` has been removed from the `eventing.knative.dev`, `messaging.knative.dev` and `flows.knative.dev` APIs. You must ensure that any existing or new resources in your deployment that are created by using these APIs use version `v1`.
+
+** The `v1alpha1` version of the `KafkaChannel` API, which was deprecated in {ServerlessProductName} version 1.14.0, has been removed. If the `ChannelTemplateSpec` of your config maps contain references to this older version, you must update this part of the spec to use the correct API version.
+
+** The deprecated `v1alpha1`, `v1alpha2`, and `v1beta1` versions of the `APIServerSource` and `SinkBinding` APIs have been removed. You must update any existing resources to use the correct API versions.
+
+** The deprecated `v1alpha2` and `v1beta1` versions of the `ContainerSource` and `PingSource` APIs have been removed. You must update any existing resources to use the correct API versions.
+
+** Version `v1beta2` of the `PingSource` API has been deprecated. References to older versions of this API should be updated to `v1` to prepare your deployment for removal of the deprecated version.
+
+[id="fixed-issues-1-17-0_{context}"]
+== Fixed issues
+
+[id="known-issues-1-17-0_{context}"]
+== Known issues
+
+* If you try to use an older version of the Knative `kn` CLI with a newer {ServerlessProductName} release, the API is not found and an error occurs.
++
+For example, if you use the 1.16.0 release of the `kn` CLI, which uses version 0.22.0, with the 1.17.0 {ServerlessProductName} release, which uses the 0.23.0 versions of the Knative Serving and Knative Eventing APIs, the CLI does not work because it continues to look for the outdated 0.22.0 API versions.
++
+Ensure that you are using the latest `kn` CLI version for your {ServerlessProductName} release to avoid issues.
+
+* If you create a new subscription for a Kafka channel, there might be a delay in the Kafka channel data plane becoming ready to dispatch messages after the newly created subscription reports a ready status. As a result, messages that are sent during the time which the data plane is not reporting a ready status might not be delivered to the subscriber.
++
+The following steps are recommended to mitigate this issue:
+** Schedule a maintenance window for updates, during which the Kafka channel is not expected to receive events.
+** If you are creating subscriptions manually, send a few test events to ensure that events are not being lost. After you have confirmed that test events are not being lost, it is safe to continue using your subscriptions as usual.
+** If your subscriptions are being created by using automation, avoid using the current version of Kafka channels included with {ServerlessProductName} 1.17.0 for mission critical or time sensitive systems where some issues being lost when subscriptions are created is unacceptable.
```

Comment:
TBH I don't think we should recommend this change for the KafkaChannel, the producer and subscriber are unaware of the kafka consumption details (consumer group names etc) . The recommendation I provided is an EDA technique (even though not optimal since it introduces a level of coupling between producers and consumers) yet it's still just another "application design choice" to abstracted from the internals of the KafkaChannel. The code would work with IMC or any other channel needed without adding code that has a strong knowledge of the implementation detail of the KafkaChannel.

Reply:
I think you misunderstood me here. I proposed to add recommendations in case automation (terraform, Ansible, user operator) is used to deploy user's app. Then user can add additional code to check if the Kafka offsets are already committed.

Reply:
Understood. I guess my suggestion was more generic to cover cases where it's a "deployment automation" or a "programmatic creation" of the subscriptions / sources. So I think it still covers this case.